### PR TITLE
DHFPROD-4173: Apply additional settings in flow runner

### PR DESCRIPTION
**Pull request by ryanjdew**
### Description

Certain settings weren't being accounted for when running a data load. Reading in those settings now. I also added the missing target format on the mapping settings form.

https://drive.google.com/open?id=1ymeBD_m5KvOnUBtLo3d_DVFFbEJFErW4

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/mappings.sjs [hunk 1]
```diff
@@ -91,5 +102,6 @@ module.exports = {
     getPermissions,
     getArtifactNode,
     validateArtifact,
-    getArtifactSettingNode
+    getArtifactSettingNode,
+    defaultArtifactSettings
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/mappings.sjs:106]: Shouldn't there be a unit test against getArtifactSettings for a mapping to verify that the defaults are returned if there aren't settings yet?

File: marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java [hunk 2]
```diff
@@ -657,17 +657,32 @@ protected void applyArtifactSettings(Step step, Map<String, Object> stepConfig)
                     if (artifactSettingsJson.hasNonNull("customHook")) {
                         step.setCustomHook(artifactSettingsJson.get("customHook"));
                     }
-                    List<String> stepOptionProperties = Arrays.asList("collections", "permissions");
                     ObjectMapper mapper = new ObjectMapper();
+                    List<String> collections = new ArrayList<>();
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java:661]: I think verifyThatLoadDataSettingsAreAppliedToStepConfig in FlowRunnerTest should be expanded to cover these fixes.

Comment (ryanjdew) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java:661]: done

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/mappings.sjs (hunk 1 above)
Comment (ryanjdew) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/mappings.sjs:106]: Added to the ML unit test to test the retrieval of default settings.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/mappings.sjs [hunk 3]
```diff
@@ -83,6 +83,17 @@ function getArtifactSettingNode(collectionName, artifactName, artifactVersion) {
   return fn.head(results);
 }
 
+function defaultArtifactSettings(artifactName) {
+    return {
+        artifactName,
+        collections: ['default-mapping'],
```
Comment (bsrikan) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/mappings.sjs:89]: @ryanjdew wondering if this should be default-mapping or entity-services-mapping?

Comment (ryanjdew) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/mappings.sjs:89]: default-mapping is the collection set by entity-services-mapping step definition.

**Comment by ryanjdew** [comment]
Updated the controller tests to expect default settings instead of empty settings.

**Comment by xnikhil08** [comment]
@ryanjdew - Just a suggestion. Not related to this PR, but the getArtifactsSettings axios call in the function getArtifactsSettings() in file activity-settings-dialog.tsx seems to be a duplicate of the similar API call present in the file utils/manageArtifacts-service.tsx(function getSettingsArtifact).
I know we have this discussion already in place to move all the API calls to the api folder in future but not sure when will that happen so until then we can use the one present in the utils folder as that was created to be used commonly across all components.

**Pull request merged**